Kacper murzyn/7.32.0 changelog (#9814)
DataDog/datadog-agent: CHANGELOG.rst

Release Notes
=============

.. _Release Notes_7.32.0:

7.32.0 / 6.32.0
======

.. _Release Notes_7.32.0_Prelude:

Prelude
-------

Release on: 2021-11-10

- Please refer to the `7.32.0 tag on integrations-core <https://github.com/DataDog/integrations-core/blob/master/AGENT_CHANGELOG.md#datadog-agent-version-7320>`_ for the list of changes on the Core Checks

.. _Release Notes_7.32.0_Upgrade Notes:

Upgrade Notes
-------------

- APM: Change default profiling intake to use v2 endpoint.

> - CSPM the check subcommand is now part of the security-agent compliance.

Reviewer: ```suggestion
- CSPM: the ``check`` subcommand is now under the ``security-agent compliance`` command (it was previously directly under ``security-agent``)
```

Author: Updated

.. _Release Notes_7.32.0_New Features:

New Features
------------

- On Kubernetes, add a `kube_priority_class` tag on metrics coming from pods with a priority class.

- Priority class name of pods are now collected and sent to the orchestration endpoint

- Autodiscovery can now resolve template variables and environment variables in log configurations.

- The Windows installer now offers US5 as a new site choice.

- APM: New telemetry was added to measure `/v.*/traces` endpoints latency and response size.
  These metrics are `datadog.trace_agent.receiver.{rate_response_bytes,serve_traces_ms}`.

- APM: Metrics are now available for Windows Pipes and UDS connections via datadog.trace_agent.receiver.{uds_connections,pipe_connections}.

- Introduce a new configuration parameter ``container_env_as_tags``
  to allow converting containerd containers' environment variables into tags.

- The "containerd" check is now supported on Windows.

- Add experimental support for writing agent-side CSPM compliance checks in Rego.

- Runtime security can now attach span/trace to event.

- Provides alternative implementation for process collection on Windows using performance counters.

> - Add multi-line auto-sensing when tailing logs from file.

Reviewer: ```suggestion
- Add multi-line auto-sensing when tailing logs from file (in beta).
```

Author: Updated

  It checks the 1000 first lines (or waits 30 seconds, whichever is first)
  when tailing for a list of known timestamp formats. If the 
  number of matched lines is greater than the threshold it 
  switches to the MultiLineHandler with the pattern matching
  the timestamp format. The pattern chosen is saved in the log

>   config and is reused if the file rotates.  Use the new global config 
>   parameter: ``logs_config.auto_multi_line_detection`` to enable
>   the feature.  New per log integration config parameters:
>   * ``auto_multi_line_detection`` to enable the feature on a case
>     by case basis

Reviewer: ```suggestion
  config and is reused if the file rotates.  Use the new global config 
  parameter ``logs_config.auto_multi_line_detection`` to enable
  the feature for the whole agent, or the per log integration config parameter ``auto_multi_line_detection``
  to enable the feature on a case by case basis.
```

  after two check runs without new samples. Rate, historate and monotonic count will be
  kept in memory for additional 25 hours after that. Number of check runs and the
  additional time can be changed with `check_sampler_bucket_commits_count_expiry` and
  `check_sampler_stateful_metric_expiration_time`. Metric expiration can be disabled
  entirely by setting `check_sampler_expire_metrics` to `false`.

- CSPM reports the agent version as part of the events

- Agents are now built with Go1.16.  This will have one user-visible change:
  on Linux, the process-level RSS metric for agent processes will be
  reduced from earlier versions.  This reflects a change in how memory
  usage is calculated, not a reduction in used memory, and is an artifact
  of the Go runtime `switching from MADV_FREE to MADV_DONTNEED
  <https://golang.org/doc/go1.16#runtime>`_.

- Tag Kubernetes containers with ``image_id`` tag.

- Eliminates the need to synchronize state between regular and RT process collection.

- APM: Added a configuration option to set the API key separately for Live
  Debugger. It can be set via `apm_config.debugger_api_key` or
  `DD_APM_DEBUGGER_API_KEY`.

> - Update EP forwarder config to use intake v2 for ndm metadata

Reviewer: ```suggestion
- Use v2 intake for NDM metadata
```

Author: Updated

> - Remove the `reason` tag from the `kubernetes_state.job.failed` metric to reduce cardinality

Reviewer: ```suggestion
- Remove the ``reason`` tag from the ``kubernetes_state.job.failed`` metric to reduce cardinality
```

Author: Updated
